shark_turbine/kernel/wave/wave.py

```diff
@@ -298,8 +319,29 @@ def test_execute(self, args, kwargs):
             )
             vm_context = ctx.vm_context
 
-            func = mod.lookup_function("isolated_benchmark")
-            _invoke(vm_context, device, func, kernel_inputs, kernel_outputs)
+            func_name = "isolated_benchmark"
+            if run_bench:
+                benchmark_flags = {}
+                if bench_batch_size is not None:
+                    benchmark_flags["batch_size"] = int(bench_batch_size)
+
+                if bench_repetitions is not None:
+                    benchmark_flags["benchmark_repetitions"] = int(bench_repetitions)
+
+                inputs = [inp.numpy() for inp in kernel_inputs]
+                benchmark_results = bench.benchmark_module(
+                    mod,
+                    entry_function=func_name,
+                    device=device,
+                    inputs=inputs,
+                    **benchmark_flags,
+                )
+                print("Benchmark results:")
+                for result in benchmark_results:
+                    print(result)
+            else:
+                func = mod.lookup_function(func_name)
```

Comment:
The other thing (also based on experience) is that often during tuning, we want to first make sure that the candidate produces correct results and only if it produces correct results do we want to benchmark it and sometimes we might want to just benchmark the function. So I would reorganize this function so that if run or bench are enabled, we do the creation of the vm context / etc. and then if only run is enabled we only run, if only bench is enabled we only bench and if both are enabled we run and bench. The idea is that we can do both run and bench through a single pass through the function which is not possible right now. What do you think?

Reply:
done

Reply:
Thanks